mono/metadata/reflection.c

```diff
@@ -12386,7 +12462,8 @@ ensure_generic_class_runtime_vtable (MonoClass *klass, MonoError *error)
 	if (!ensure_runtime_vtable (gklass, error))
 		return FALSE;
 
-	fix_partial_generic_class (klass);
+	fix_partial_generic_class (klass, error);
+	return_val_if_nok (error, FALSE);
```

Comment:
fix_partial_generic_class cleanup the error. We should not poke into one after that.